clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -664,14 +663,15 @@ public void unmuteAll() {
             unmute(channel);
     }
 
-    private void completeDelayedChannelClose() throws IOException {
+    private void completeDelayedChannelClose() {
         if (delayedClosingChannels == null)
             return;
 
         DelayedChannelClose delayedClose;
-        while (((delayedClose = delayedClosingChannels.peek()) != null) && delayedClose.isCloseable()) {
-            handleCloseOnPrepareFailure(delayedClose.channel, delayedClose.closeMode);
-            delayedClosingChannels.remove();
+        if (delayedClosingChannels.size() > 0) {
+            long now = System.currentTimeMillis();
```

Comment:
Just so that I understand, is the reason you suggest using `Time#milliseconds()` so that we could test this using `MockTime`?

Reply:
Yeah, we should in general always use `Time` in the code so that we can override it's behaviour when needed. The second thing is that it's useful to avoid unnecessary calls as they can be a bit expensive depending on the context.